### ISSUE TITLE
Class of classmethod does not support ``__orig_bases__`` while Type does

### ISSUE THREAD

**Issue by prauscher** [pylint-dev/pylint#7995]
### Bug description

``__orig_bases__`` is defined in [PEP 560](https://peps.python.org/pep-0560/#mro-entries).

```python
# pylint: disable=missing-docstring,too-few-public-methods
from typing import Generic, TypeVar, Type

T = TypeVar("T")

class Base(Generic[T]):
    @classmethod
    def get_type(cls) -> Type:
        # the following line yields a no-member error for __orig_bases__
        return cls.__orig_bases__[0].__args__[0]


class SubClass(Base[int]):
    pass


def get_type(cls: Type) -> Type:
    return cls.__orig_bases__[0]


# all three lines give the correct result <class 'int'>
print(SubClass.get_type())
# the following line yields a no-member error for __orig_bases__
print(SubClass.__orig_bases__[0].__args__[0])
# the following line yields no error
print(get_type(SubClass).__args__[0])
```

### Command used

```shell
pylint test11.py
```

### Pylint output

```shell
************* Module test11
test11.py:11:15: E1101: Class 'Base' has no '__orig_bases__' member (no-member)
test11.py:25:6: E1101: Class 'SubClass' has no '__orig_bases__' member (no-member)
```

### Expected behavior

(no error)

### Pylint version

```shell
pylint 2.15.0
astroid 2.12.4
Python 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0]
```

### OS / Environment

Arch Linux
